src/reconnection.ts

```diff
@@ -61,23 +62,52 @@ export class ReconnectManager<PeerMetadata, TrackMetadata> {
     this.connect = connect;
     this.reconnectConfig = createReconnectConfig(config);
 
-    this.client.on('socketError', () => {
+    const socketError: MessageEvents<
```

Comment:
nit: maybe rename it to `onSocketError`? so it's suggesting it's a funciton, not an error (the same applies below)